.storybook/stories/DataFiltering/components/DashboardPagination.tsx

```diff
@@ -24,8 +24,37 @@ export const DashboardPagination = () => {
 			})}
 			itemsPerPage={pagination.perPage}
 			onItemsPerPageChange={(perPage) =>
-				setPagination({ ...pagination, perPage })
+				setPagination({
+					page: getValidPage(
+						pagination.page,
+						totalItems,
+						perPage,
+						pagination.perPage
+					),
+					perPage,
+				})
 			}
 		/>
 	);
 };
+
+const getValidPage = (
+	currentPage: number,
+	totalItems: number,
+	perPage: number,
+	prevPerPage: number
```

Comment:
I think using an object as the arg like for `generatePaginationRangeText` on L18 will make it easier to understand what each part is

Reply:
- [ ] objectify the args